### PR TITLE
Pass role prop to List & ListItem

### PR DESCRIPTION
**Pull request by epilande**
#### What does this PR do?

Pass role prop to List & ListItem component.
Update List component to use classnames & es6 updates.

### CODE REVIEW

File: src/js/components/List.js [hunk 1]
```diff
@@ -98,38 +99,51 @@ export default class List extends Component {
...
     }
 
     return (
-      <ul ref={(ref) => this.listRef = ref}
-        className={classes.join(' ')} onClick={this._onClick}>
+      <ul
+        ref={(ref) => this.listRef = ref}
+        className={classes}
+        onClick={this._onClick}
+        role={role}
```
Comment (alansouzati) [inline src/js/components/List.js]: I don't think we need to explicitly declare "role" here, but transfer the props down to the ul component.

**Box implementation for transferring props**: https://github.com/grommet/grommet/blob/master/src/js/components/Box.js#L72

**Original issue tracking the problem**:

https://github.com/grommet/grommet/issues/85

File: src/js/components/List.js [hunk 2]
```diff
@@ -149,3 +163,7 @@ List.propTypes = {
     PropTypes.arrayOf(PropTypes.number)
   ])
 };
+
+List.defaultProps = {
+  role: 'list'
```
Comment (alansouzati) [inline src/js/components/List.js]: I like the idea of having a default prop for role. but we still need the transfer props. 

I'm ok with leaving role prop if we decide to keep default props.

File: src/js/components/ListItem.js [hunk 3]
```diff
@@ -47,5 +47,6 @@ ListItem.defaultProps = {
   align: 'center',
   direction: 'row',
   pad: {horizontal: 'medium', vertical: 'small'},
-  separator: 'bottom'
+  separator: 'bottom',
+  role: 'listitem'
```
Comment (alansouzati) [inline src/js/components/ListItem.js]: I like the idea of having a default prop for role. but we still need the transfer props. 

I'm ok with leaving role prop if we decide to keep default props.

Comment (epilande) [inline src/js/components/ListItem.js]: ListItem is transferring `{...boxProps}` which includes role.

**Pull request merged**